activities.py

```diff
@@ -0,0 +1,72 @@
+# @@@SNIPSTART python-money-transfer-project-template-withdraw
+from temporalio import activity
+
+from banking_service import BankingService, InvalidAccountError
+from shared import PaymentDetails
+
+
+# Generic errors for activity failures
+class WithdrawFailedError(Exception):
+    pass
+
+
+class DepositFailedError(Exception):
+    pass
+
+
+class RefundFailedError(Exception):
+    pass
+
+
+@activity.defn
+async def withdraw(data: PaymentDetails) -> str:
+    reference_id: str = f"{data.reference_id}-withdrawal"
+    bank = BankingService("bank-api.example.com")
+    try:
+        confirmation = bank.withdraw(data.source_account, data.amount, reference_id)
+        return confirmation
+    except InvalidAccountError as error:
+        raise error
+    except Exception as error:
+        activity.logger.error(f"Withdrawal failed: {error}")
```

Comment:
```suggestion
        activity.logger.exception(f"Withdrawal failed: {error}")
```

If you want the exception. Though we already log these at the warn level IIRC.